src/LinkedList.js

```diff
@@ -0,0 +1,129 @@
+class Node {
+  constructor() {
+    this.data
+    this.next = null
+  }
+}
+
+class LinkedList {
+  constructor() {
+    this.head = null
+    this.tail = null
+  }
+
+  getHeadNode() {
+    return this.head
+  }
+
+  getTailNode() {
+    return this.tail
+  }
+
+  contains(element) {
+    let elementFound = false
+    let iterator = this.head
+    while(iterator) {
+      if(iterator.data === element) elementFound = true
+      iterator = iterator.next
+    }
+    return elementFound
+  }
+
+  find(element) {
+    let iterator = this.head
+    while(iterator) {
+      if(iterator.data === element) return iterator
+      iterator = iterator.next
+    }
+    return -1
+  }
+
+  insert(element) {
+    let tail = new Node()
+    tail.data = element
+    if(this.head == null) this.head = tail
+    if(this.tail != null) this.tail.next = tail
+
+    this.tail = tail
+
+  }
+
+  insertFirst(element) {
+    let head = new Node()
+    head.data = element
+    head.next = this.head
+    this.head = head
+  }
+
+  insertBefore(match,element) {
+    let insertNode = new Node()
+    insertNode.data = element
+
+    let matchNode = this.head
+    while(matchNode.data !== match && matchNode) matchNode = matchNode.next
```

Comment:
Reverse the order of these. You want a `matchNode` to exist before looking at `matchNode.data`.